pkg/api/resourcequota/util.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resourcequota
+
+import (
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	api "k8s.io/kubernetes/pkg/apis/core"
+	"k8s.io/kubernetes/pkg/features"
+)
+
+// DropDisabledFields removes disabled fields from the ResourceQuota spec.
+// This should be called from PrepareForCreate/PrepareForUpdate for all resources
+// containing a ResourceQuota spec.
+func DropDisabledFields(newResSpec *api.ResourceQuotaSpec, oldResSpec *api.ResourceQuotaSpec) {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.PodAffinityNamespaceSelector) &&
+		newResSpec != nil && newResSpec.Scopes != nil &&
+		!crossNamespaceScopeInUse(oldResSpec) {
+		scopes := []api.ResourceQuotaScope{}
```

Comment:
I want it initialized.

Reply:
why? The rest of the code should be able to handle a nil slice.

Reply:
not exactly, I don't want to set newResSpec.Scopes to nil if it was empty; I modified the condition to test for `len` instead of nil so this would work.

Reply:
you shouldn't distinguish between nil and empty slice. Where do you need that distinction?

Reply:
I guess some existing tests rely on empty slice.

Reply:
that is not the place were we decide whether or not there should be a distinction, the function should simply drop the value if it exists, and not change anything else.